File: elasticsearches.js
#!/usr/bin/env node

'use strict';
var http = require('http');
var url = require('url');

var marathonUrl = process.argv[2];
var appId = process.argv[3];

if  (marathonUrl && appId) {
	var taskUrl = marathonUrl + '/v2/apps/' + appId + '/tasks'
	var parsedUrl = url.parse(taskUrl);

	var options = {
	  hostname: parsedUrl.hostname,
	  port: parsedUrl.port,
	  path: parsedUrl.path,
	  method: 'GET',
	  headers: {
		'Content-Type': 'application/json',
		'Accept': 'application/json'
	  }
	};

	http.get(options, function(res) {
	  var data = '';
	  res.on('data', function(chunk) {
	    data += chunk;
	  });
	  res.on('end', function(chunk) {
	    var apps = JSON.parse(data.toString()).tasks;
		var output = [];
		apps.forEach(function(element, index, array) {
			output.push(element.host + ':' + element.ports[0]);
		});
		console.log(output.join(','));
	  });
	});
}

File: elasticsearches.rb
#!/usr/bin/env ruby

require 'net/http'
require 'json'
require 'uri'

marathon_url = ARGV[0]
app_id = ARGV[1]

tasks_url = "#{marathon_url}/v2/apps/#{app_id}/tasks"

uri = URI.parse(tasks_url)
http = Net::HTTP.new(uri.host, uri.port)

request = Net::HTTP::Get.new(uri.path)
request['Content-Type'] = 'application/json'
request['Accept'] = 'application/json'

response = http.request(request)

data = JSON.parse(response.body)
print data['tasks'].map { |t| "#{t['host']}:#{t['ports'].first}" }.join(',')
